--- tmp/jboss-4.0.5.GA/bin/run.bat  2006-10-23 17:59:26.000000000 -0400
+++ java/jboss-4.0.5.GA/bin/run.bat 2010-07-23 15:16:12.026921600 -0400
@@ -63,11 +63,8 @@
 rem If JBOSS_CLASSPATH is empty, don't include it, as this will 
 rem result in including the local directory, which makes error tracking
 rem harder.
-if "%JBOSS_CLASSPATH%" == "" (
-  set JBOSS_CLASSPATH=%JAVAC_JAR%;%RUNJAR%
-) else (
-  set JBOSS_CLASSPATH=%JBOSS_CLASSPATH%;%JAVAC_JAR%;%RUNJAR%
-)
+if not "%JBOSS_CLASSPATH%" == "" set if JBOSS_CLASSPATH=%JBOSS_CLASSPATH%;
+set JBOSS_CLASSPATH=%JBOSS_CLASSPATH%%JAVAC_JAR%;%RUNJAR%
 
 rem Setup JBoss specific properties
 set JAVA_OPTS=%JAVA_OPTS% -Dprogram.name=%PROGNAME%
@@ -78,7 +75,7 @@
 if not errorlevel == 1 (set JAVA_OPTS=%JAVA_OPTS% -server)
 
 rem JVM memory allocation pool parameters. Modify as appropriate.
-set JAVA_OPTS=%JAVA_OPTS% -Xms128m -Xmx512m
+set JAVA_OPTS=%JAVA_OPTS% "-Xbootclasspath/a:%JBOSS_HOME%\lib\nexj-boot.jar" -Xms128m -Xmx512m -XX:MaxPermSize=128M
 
 rem With Sun JVMs reduce the RMI GCs to once per hour
 set JAVA_OPTS=%JAVA_OPTS% -Dsun.rmi.dgc.client.gcInterval=3600000 -Dsun.rmi.dgc.server.gcInterval=3600000
@@ -86,6 +83,13 @@
 rem JPDA options. Uncomment and modify as appropriate to enable remote debugging.
 rem set JAVA_OPTS=-Xdebug -Xrunjdwp:transport=dt_socket,address=8787,server=y,suspend=y %JAVA_OPTS%
 
+rem Eclipse Profiler options for CPU profiling
+rem set JAVA_OPTS=-noverify -XrunProfilerDLL:1 -Xbootclasspath/a:jakarta-regexp.jar;profiler_trace.jar;commons-lang.jar -D__PROFILER_PACKAGE_FILTER=__A__org.jboss.Main;__M__sun.;__M__com.sun.;__M__java.;__M__javax. -D__PROFILER_TIMING_METHOD=1 %JAVA_OPTS%
+
+rem Eclipse Profiler options for heap profiling
+rem set JAVA_OPTS=-noverify -XrunProfilerDLL:3,10,0 -D__PROFILER_PROFILE_HEAP=1 -Xbootclasspath/a:jakarta-regexp.jar;profiler_trace.jar;commons-lang.jar -D__PROFILER_PACKAGE_FILTER=__A__org.jboss.Main;__M__ -D__PROFILER_TIMING_METHOD=1 %JAVA_OPTS%
+
+
 rem Setup the java endorsed dirs
 set JBOSS_ENDORSED_DIRS=%JBOSS_HOME%\lib\endorsed
 
